### PR TITLE
fix(text-input): remove outline flicker during state transition

### LINKED ISSUE #5979: [Text input] Transient gray border shown when transitioning from disabled to enabled
## What package(s) are you using?

- [x] `carbon-components`
- [ ] `carbon-components-react`

## Detailed description

This can be reproduced using the React playground: https://react.carbondesignsystem.com/?path=/story/textinput--default. The bug can be seen in both Chrome and FireFox.

image

Simply change the text input from disabled to enabled. To make it easier to see the transition, set the animation duration to a large value:

image

The text input component has a CSS transition for outline property: https://github.com/carbon-design-system/carbon/blob/f107c20fa1f661f6b1dd435c2ca289a8ae687978/packages/components/src/components/text-input/_text-input.scss#L35

For enabled state, the outline is `2px solid transparent`; for disabled state, the outline is `none`. It seems the transient gray border is shown when transitioning from `none` to `2px solid transparent`. We probably can fix it by using `2px solid transparent` for the disabled state too.

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/5979

Sets outline to transparent when disabled so that there is not a visible state change when toggling between disabled/enabled. 

#### Changelog

**Changed**

- Instead of setting the outline to `none`, we set it to transparent. 

#### Testing / Reviewing

Check out the `TextInput` story, and toggle the disabled state a few times. You should not see any flicker around the input.

### CODE REVIEW

File: packages/components/src/components/text-input/_text-input.scss [hunk 1]
```diff
@@ -142,7 +142,7 @@
   //-----------------------------
   .#{$prefix}--text-input:disabled {
     cursor: not-allowed;
-    outline: none;
+    outline: 2px solid transparent;
```
Comment (yinglejia) [inline packages/components/src/components/text-input/_text-input.scss]: Use `@include focus-outline('reset');` instead?

Comment (tw15egan) [reply packages/components/src/components/text-input/_text-input.scss]: Yes! 🤦 

Updated, thanks for catching that.

**Review by emyarod — APPROVED** [review]
looks good to me pending visual review

**Review by laurenmrice — APPROVED** [review]
looks good to me !

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @tw15egan!

**Review by joshblack — APPROVED** [review]

**Pull request merged**